pkgs/applications/window-managers/sway/wrapper.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, sway-unwrapped
+, swaybg
+, dbus
+, wrapGAppsHook
+, symlinkJoin
+, writeShellScriptBin
+, withBaseWrapper ? true
+, withGTKWrapper ? false
+, extraSessionCommands ? ""
+}:
+
+with lib;
+
+let
+  swayPackage = sway-unwrapped;
+
+  baseWrapper = writeShellScriptBin "sway" ''
+    set -o errexit
+
+    if [ ! "$_SWAY_WRAPPER_ALREADY_EXECUTED" ]; then
+      export _SWAY_WRAPPER_ALREADY_EXECUTED=1
+      ${extraSessionCommands}
+    fi
+
+    if [ "$DBUS_SESSION_BUS_ADDRESS" ]; then
+      export DBUS_SESSION_BUS_ADDRESS
+      exec ${swayPackage}/bin/sway "$@"
+    else
+      exec ${dbus}/bin/dbus-run-session ${swayPackage}/bin/sway "$@"
+    fi
+  '';
+
+  swayJoined = symlinkJoin {
+    name = "sway-base-wrapped";
+    paths = (optional withBaseWrapper [baseWrapper]) ++ [swayPackage];
+  };
+in
+symlinkJoin {
+  name = "sway";
+
+  paths = [swayJoined];
+
+  version = sway-unwrapped.version;
+
+  buildInputs = []
+    ++ (optional withGTKWrapper [wrapGAppsHook])
+    ++ sway-unwrapped.buildInputs;
```

Comment:
This shouldn't be necessary. Is it for other reasons (e.g. to easily get Sway's build dependencies, though we could get them though `sway-unwrapped`)?

Reply:
Like for the `postBuild` hook, if I drop the `sway-unwrapped.buildInputs` the GTK Wrapper seems to miss some variables and `gsettings` seems to stop working again.

Reply:
@primeos I've just checked https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/wrap-gapps-hook.sh and it seems it uses the variable present at build time to reexport them in a wrapper.
I might be compltely wrong, but since I use a `symlinkJoin`, could it be that the environment exposed while creating the wrapper is not the same, thus, it's not wrapping correctly `sway-unwrapped` ?